Entity/CustomItem.php

```diff
@@ -611,4 +620,39 @@ public function getRelationsByType(string $entityType): Collection
                 return new ArrayCollection([]);
         }
     }
+
+    public function getUniqueHash(): ?string
+    {
+        return $this->uniqueHash;
+    }
+
+    public function setUniqueHash(?string $uniqueHash): void
+    {
+        $this->uniqueHash = $uniqueHash;
+    }
+
+    public function createUniqueHash(): ?string
+    {
+        $rowData                = [];
+        $uniqueHash             = [];
+        $uniqueIdentifierFields = $this->customObject->getFieldsIsUniqueIdentifier();
+
+        if (is_null($uniqueIdentifierFields)) {
```

Comment:
Are you sure that this can be null? This is how ArrayCollection define the `filter()` method:
```
/**
     * {@inheritDoc}
     *
     * @return static
     * @psalm-return static<TKey,T>
     */
    public function filter(Closure $p)
    {
        return $this->createFrom(array_filter($this->elements, $p, ARRAY_FILTER_USE_BOTH));
    }
```

So I'd expect
```suggestion
        if (0 === $uniqueIdentifierFields->count())) {
```

Reply:
Aahh yes, good catch. I wrote this as a rough first draft for this method and didn't think about this later. I'll change this.